### DISCUSSION TITLE
LTS Schedule

### DISCUSSION THREAD

**Discussion by MikeBauerCA** [npm/feedback#204] [CLI] [answered]
On nodejs.org there is a [release schedule](https://nodejs.org/en/about/releases/) providing visibility into when certain major versions reach LTS. Is there a similar schedule for NPM CLI? Since NPM CLI v7 has been tagged as LTS, will it be included in the LTS download of Node at https://nodejs.org/en/?

On a separate note, it looks like the [doc](https://docs.npmjs.com/cli/v6/commands) needs to be updated to reflect the new LTS version. 
![image](https://user-images.githubusercontent.com/35504792/107265718-9c598480-6a12-11eb-9058-ab6760afae57.png)

Thank you all for continuing to progress this technology.

**Comment by MylesBorins** [comment, marked as answer]
update: the consensus is that npm 7 is not LTS. Further, in referencing our [Support Policy](https://github.com/npm/cli/wiki/Support-Policy) we don’t actually have an LTS program at npm, it is an artifact of the pre npm 4 release. What we do guarantee is that we will do active development on the latest major release and that we will do security patches for all releases shipped with maintained versions of Nodejs.

With this in mind we were thinking it would make sense to remove the lts dist-tag and reference from our CLI docs all together, but need to do a bit of research first to determine if there will be unexpected failures with removing the tag.

--

@MikeBauerCA can you point me towards where npm 7 is being considered LTS? It was recently promoted to be Generally available but is not yet considered LTS. We are working with the Node.js release team, of which I am a member, to figure out if we can backport npm 7 to Node.js 14, but tbh it may not be able to happen due to the various breaking changes.

As for our LTS policy, I thought we had something documented but can't find it right this moment. But TL;DR we will continue to do maintenance support to versions of npm that are being shipping with maintained versions of Node.js.

**Reply by MikeBauerCA** [reply]
Thank you. At https://www.npmjs.com/package/npm under versions, currently v7.5.3 is tagged as `lts`.
![image](https://user-images.githubusercontent.com/35504792/107366407-1f7ae900-6aac-11eb-99fb-a71a29ed8dde.png)


Similarly, this is the output of `npm view npm`
![image](https://user-images.githubusercontent.com/35504792/107366870-a039e500-6aac-11eb-84b2-0a0fbe043017.png)

**Reply by t1m0thyj** [reply]
> We are working with the Node.js release team, of which I am a member, to figure out if we can backport npm 7 to Node.js 14, but tbh it may not be able to happen due to the various breaking changes.

Thanks for your work on NPM 7.x - I'm excited about new features like workspaces!

If you do determine it's possible to backport npm@7 to Node.js 14, please consider delaying this until npm@7 is more mature and stable. ATM it has some issues that break the behavior of npm@6 and my team is unable to support customers who use npm@7 (https://github.com/npm/cli/issues/2659, https://github.com/npm/cli/issues/2660).

**Reply by MikeBauerCA** [reply]
@MylesBorins could you inform whether the promotion of v7 to lts on the registry was intentional?

**Reply by MylesBorins** [reply]
Hey Mike, I managed to find time to chat with the team and the consensus is that npm 7 is not LTS. Further, in referencing our [Support Policy](https://github.com/npm/cli/wiki/Support-Policy) we don’t actually have an LTS program at npm, it is an artifact of the pre npm 4 release. What we do guarantee is that we will do active development on the latest major release and that we will do security patches for all releases shipped with maintained versions of Nodejs.

With this in mind we were thinking it would make sense to remove the `lts` dist-tag and reference from our CLI docs all together, but need to do a bit of research first to determine if there will be unexpected failures with removing the tag.

**Reply by MikeBauerCA** [reply]
Thanks @MylesBorins. The `lts` distribution tag is what caused my confusion. I also hadn't found the Support Policy you referenced. Thanks for sharing. Could you recommend the best way to stay informed of target dates for moving versions of NPM CLI from `next` to `latest` and for when new major versions of NPM CLI may be included in Node LTS distributions?

**Reply by darcyclarke** [reply]
Hey @MikeBauerCA! Apologize for the confusion; The `lts` tag, & corresponding docs references, should be removed by **February 28th** (there has been a note/update to our [Support Policy](https://github.com/npm/cli/wiki/Support-Policy#long-term-support-lts) outlining that). 

In terms of `next`/`latest` - our team has changed its release cadence from bi-weekly &/or monthly to weekly & so we've been setting `next` at the same time as `latest`; This is partly because our increased speed has made maintaining/shuffling multiple dist-tags more tedious & less valuable. That said, it doesn't mean we won't go back to using that tag again in a more meaningful way in the future, especially as we start queuing up larger features or if we slow down the release cadence, later into the life of v7. `next` has been historically used as a pre-release channel which definitely has a lot of value.

In terms of the CLI in Node LTS releases, v7 currently ships in Node 15 with the hope/expectation is that it will be shipping with Node 16 as well - I think there are ongoing discussions on whether to backport it to Node 14 but currently, v6 is still what is shipping, by default, in Node 12/14. Of note, we actively open PRs against Node when we cut new releases of the CLI. 

Hopefully, that gives you some more insight into how we're working/queuing up work today
